util/topgen/intermodule.py

```diff
@@ -390,6 +387,54 @@ def check_intermodule_field(obj: OrderedDict, prefix: str = "") -> int:
     return error
 
 
+def find_otherside_modules(topcfg: OrderedDict, m,
+                           s) -> List[Tuple[str, str, str]]:
+    """Find far-end port based on given module and signal name
+    """
+    signame = "{}.{}".format(m, s)
+    for req, rsps in topcfg["inter_module"]["connect"].items():
+        if req.startswith(signame):
+            # return rsps after splitting module instance name and the port
+            result = []
+            for rsp in rsps:
+                rsp_m, rsp_s, rsp_i = filter_index(rsp)
+                result.append(('connect', rsp_m, rsp_s))
+            return result
+
+        for rsp in rsps:
+            if signame == rsp:
+                req_m, req_s, req_i = filter_index(req)
+                return [('connect', req_m, req_s)]
+
+    # If reaches here, no matching results in 'connect'
+    # so search 'top' or 'external'
+
+    # todo: something here
+    for sig in topcfg["inter_signal"]["signals"]:
+        # Special Handling for custom connect
+        if m == "main" and s == "tl_corei":
+            return [('top', 'rv_core_ibex', 'tl_i')]
+
+        if m == "main" and s == "tl_cored":
+            return [('top', 'rv_core_ibex', 'tl_d')]
+
+        if m == "main" and s == "tl_dm_sba":
+            return [('top', 'dm_top', 'tl_h')]
+
+        if m == "main" and s == "tl_debug_mem":
+            return [('top', 'dm_top', 'tl_d')]
+
+        # If not falls into the special category, no way to find other side
+
+        if sig["inst_name"] == m and sig["name"] == s:
+            return [('top', '', sig["top_signame"])]
+
+    # if reaches here, it means either the format is wrong, or floating port.
+    log.error("`find_otherside_modules()`: "
+              "No such signal {}.{} exists.".format(m, s))
+    return [('', '', '')]
```

Comment:
Is this better than returning something like `None` or an empty list? A list of empty strings seems rather odd to me.

Reply:
Empty list is much better. Will revise